nuxeo-core/nuxeo-platform-commandline-executor/src/main/java/org/nuxeo/ecm/platform/commandline/executor/service/executors/ShellExecutor.java

```diff
@@ -190,18 +182,12 @@ protected ExecResult exec1(CommandLineDescriptor cmdDesc, CmdParameters params,
      * @since 7.10
      */
     public static Thread pipe(InputStream in, OutputStream out) {
-        Runnable run = new Runnable() {
-            @Override
-            public void run() {
-                try {
-                    IOUtils.copy(in, out);
-                    out.flush();
-                } catch (IOException e) {
-                    throw new RuntimeException(e);
-                } finally {
-                    IOUtils.closeQuietly(in);
-                    IOUtils.closeQuietly(out);
-                }
+        Runnable run = () -> {
+            try (in; out) {
+                IOUtils.copy(in, out);
+                out.flush();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
```

Comment:
Why can we remove the `finally` block, InputStream and OutputStream not being AutoCloseable?

Reply:
In fact we can because I put them in the try-with-ressource block, see https://github.com/nuxeo/nuxeo/pull/3198/files#diff-f16663790aa29f66f97dee134cd6ec7fR176

Reply:
:+1:, just missed that `Closeable` was extending `AutoCloseable`.